io/serialport/25-serial.js

```diff
@@ -200,6 +224,42 @@ module.exports = function(RED) {
     }
     RED.nodes.registerType("serial request", SerialRequestNode);
 
+
+    // Serial Control Node
+    function PortSelectNode(n) {
+        RED.nodes.createNode(this,n);
+        this.serialConfig = RED.nodes.getNode(n.serial);
+
+        if (!this.serialConfig) {
+            this.error(RED._("serial.errors.missing-conf"), {});
+            return;
+        }
+
+        this.serial = n.serial;
+        var node = this;
+        node.port = serialPool.get(this.serialConfig);
+        node.on("input",function(msg) {
+            if (msg.payload.hasOwnProperty("config")) { 
```

Comment:
I think this is now too complicated...  I think whatever the input is, the node should just reply with the (new) config - I.e. the current config if nothing changes - or the updated one if something has changed. No need for a config property - just the payload as you had it before.... (but see below)